packages/accordions/src/elements/stepper/Stepper.tsx

```diff
@@ -5,33 +5,46 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { useRef, useEffect, forwardRef, useMemo } from 'react';
+import React, { Children, forwardRef, useMemo } from 'react';
 import { IStepperProps } from '../../types';
 import { StyledStepper } from '../../styled';
-import { StepperContext } from '../../utils';
+import { StepContext, StepperContext } from '../../utils';
 import { Step } from './components/Step';
 import { Label } from './components/Label';
 import { Content } from './components/Content';
 
 const StepperComponent = forwardRef<HTMLOListElement, IStepperProps>(
-  ({ isHorizontal, activeIndex, ...props }, ref) => {
-    const currentIndexRef = useRef(0);
+  ({ isHorizontal, activeIndex, children, ...props }, ref) => {
     const stepperContext = useMemo(
       () => ({
         isHorizontal: isHorizontal || false,
-        activeIndex: activeIndex!,
-        currentIndexRef
+        activeIndex: activeIndex!
       }),
-      [isHorizontal, activeIndex, currentIndexRef]
+      [isHorizontal, activeIndex]
     );
 
-    useEffect(() => {
-      currentIndexRef.current = 0;
-    });
-
     return (
       <StepperContext.Provider value={stepperContext}>
-        <StyledStepper ref={ref} isHorizontal={isHorizontal} {...props} />
+        <StyledStepper ref={ref} isHorizontal={isHorizontal} {...props}>
+          {useMemo(
+            () =>
+              Children.toArray(children).map((child, index) => (
+                <StepContext.Provider
+                  key={index}
+                  // eslint-disable-next-line react/jsx-no-constructed-context-values
```

Comment:
Could you extract `const value: IStepContext = { ... };` to the top of this function to get rid of this `eslint-disable`?

Reply:
Moving the variable doesn't get rid of the `eslint-disable` rule unfortunately. The rule states that context value should be wrapped in a `useMemo` to avoid re-renders- being that it is already in a `useMemo` it's not permitted nor does it give any benefit.